src/acl/external/unix_group/check_group.cc

```diff
@@ -170,11 +170,17 @@ main(int argc, char *argv[])
         case 'p':
             check_pw = 1;
             break;
-        case 'g':
-            grents = (char**)realloc(grents, sizeof(*grents) * (ngroups+1));
+        case 'g': {
+            char **tmpgrents = (char**)realloc(grents, sizeof(*grents) * (ngroups+1));
+            if (!tmpgrents) {
+                fprintf(stderr, "FATAL: memory allocation failed\n");
+                exit(EXIT_FAILURE);
+            }
+            grents = tmpgrents;
```

Comment:
This should be using the portable libcompat API ```xrealloc()``` which handles any errors internally and guarantees to only return on success.

```suggestion
        case 'g':
            char **tmpgrents = (char**)xrealloc(grents, sizeof(*grents) * (ngroups+1));
            grents = tmpgrents;
```